### PR TITLE
LOK-2462:  Node Inventory Table: Sorting and pagination

### PR DESCRIPTION
**Pull request by Syed-Bilal-Haider-Dataq**
## Description
Implement sorting and pagination on Node Inventory table.-->

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2462

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: ui/src/components/Inventory/InventoryFilter.vue [hunk 1]
```diff
@@ -2,14 +2,14 @@
   <ul class="filter-container">
     <li v-if="!onlyTags" class="autocomplete flex margin-bottom">
       <FeatherSelect class="filter-type-selector" label="Search Type" textProp="name"
-        :options="[{ id: 1, name: 'Labels' }, { id: 2, name: 'Tags' }]" :modelValue="inventoryStore.searchType"
+        :options="[{ id: 1, name: 'label' }, { id: 2, name: 'Tag' }]" :modelValue="inventoryStore.searchType"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryFilter.vue:5]: Create a `Comparator` for `options` (see `monitoringPolicies.constants.ts`). Keep the old names and don't send them to graphQL query send `id`

File: ui/src/components/Inventory/InventoryFilter.vue [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
         @update:modelValue="inventoryStore.setSearchType">
       </FeatherSelect>
-      <FeatherAutocomplete v-if="inventoryStore.searchType.name === 'Tags'" type="multi"
+      <FeatherAutocomplete v-if="inventoryStore.searchType.name === 'Tag'" type="multi"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryFilter.vue:8]: See first comment and after creating `Comparator` use that here. Move each prop in its own line for better readability

File: ui/src/components/Inventory/InventoryFilter.vue [hunk 3]
```diff
@@ -2,14 +2,14 @@
...
         :modelValue="inventoryStore.tagsSelected" @update:modelValue="searchNodesByTags"
         :results="tagQueries.tagsSearched" @search="tagQueries.getTagsSearch" class="inventory-auto" label="Search"
         :allow-new="false" textProp="name" render-type="multi" data-test="search-by-tags" ref="searchNodesByTagsRef" />
-      <FeatherInput v-if="inventoryStore.searchType.name === 'Labels'" @update:model-value="searchNodesByLabel"
+      <FeatherInput v-if="inventoryStore.searchType.name === 'label'" @update:model-value="searchNodesByLabel"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryFilter.vue:12]: See first comment and after creating `Comparator` use that here. Move each prop in its own line for better readability

File: ui/src/components/Inventory/InventoryTabTable.vue [hunk 4]
```diff
@@ -62,29 +85,58 @@ defineProps({
...
+  { id: 'nodeAlias', label: 'Node Name' },
+  { id: 'nodeLabel', label: 'Management IP' }
 ]
 
+const page = computed(() => store.inventoryNodesPagination.page || 1)
+const pageSize = computed(() => store.inventoryNodesPagination.pageSize)
+const total = computed(() => store.inventoryNodesPagination.total)
+const data = computed(() => store.nodes || [])
+
+const hasAlerts = computed(() => {
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabTable.vue:107]: Rename this variable to `hasNodes`

File: ui/src/store/Views/inventoryStore.ts [hunk 5]
```diff
@@ -16,17 +31,37 @@ export const useInventoryStore = defineStore('inventoryStore', {
     unmonitoredFilterActive: false,
     detectedFilterActive: false,
     nodesSelected: [] as NewInventoryNode[],
-    searchType: { id: 1, name: 'Labels' },
+    searchType: { id: 1, name: 'label' },
```
Comment (Shahbaz-dataq) [inline ui/src/store/Views/inventoryStore.ts:34]: See first comment and after creating `Comparator` use that here.

File: ui/src/store/Views/inventoryStore.ts [hunk 6]
```diff
@@ -87,9 +122,48 @@ export const useInventoryStore = defineStore('inventoryStore', {
     },
     setSearchType(searchType: { id: number, name: string }) {
```
Comment (Shahbaz-dataq) [inline ui/src/store/Views/inventoryStore.ts:123]: Remove the destructure object its node being used

File: ui/src/components/Inventory/InventoryTabTable.vue [hunk 7]
```diff
@@ -62,29 +85,58 @@ defineProps({
   }
 })
 
+const emptyListContent = {
+  msg: 'No results found.'
+}
+
 const sort = reactive({
   NodeName: SORT.ASCENDING,
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabTable.vue:93]: Set sorting for `NodeName` to `Sort.None`. Initially nodes are being sorted on `id`

File: ui/src/components/Inventory/InventoryTabTable.vue [hunk 8]
```diff
@@ -62,29 +85,58 @@ defineProps({
...
+})
+
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const sortChanged = (sortObj: Record<string, string>) => {
-  console.log('sort inventory')
+const sortChanged = (columnId: string, sortObj: Record<string, string>) => {
+  startSpinner()
+  if (sortObj.value === 'asc' || sortObj.value === 'desc') {
+    const sortAscending = sortObj.value === 'asc'
+    const sortByAlerts = { sortAscending, sortBy: columnId }
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabTable.vue:116]: Rename this variable to something like `sortByNode` declare it after `startSpinner()` initialize it in `if` and `else`

File: ui/src/components/Inventory/InventoryFilter.vue [hunk 9]
```diff
@@ -2,14 +2,14 @@
   <ul class="filter-container">
     <li v-if="!onlyTags" class="autocomplete flex margin-bottom">
       <FeatherSelect class="filter-type-selector" label="Search Type" textProp="name"
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryFilter.vue:4]: Move each prop in its own line for better readability

File: ui/src/components/Inventory/InventoryTabTable.vue [hunk 10]
```diff
@@ -38,6 +43,20 @@
             </tr>
           </TransitionGroup>
         </table>
+        <div v-if="!hasAlerts || total === 0">
```
Comment (Shahbaz-dataq) [inline ui/src/components/Inventory/InventoryTabTable.vue:46]: Rename this variable `hasAlerts`

**Review by Shahbaz-dataq — CHANGES_REQUESTED** [review]
Other than these changes their is a major problem. After updating `BuildNetworkInventoryPageDocument` to support pagination and sorting. This same document is being used for `Card` Tab content for nodes. Due to no pagination on `Card` tab only 10 nodes will be displayed initially. And their is no way to change the page until you go to `Table` Tab change page and come back to `Card` Tab. 
My suggestion apply pagination on `Card` Tab as well just as it is in `Table` Tab.

File: ui/src/store/Views/inventoryStore.ts [hunk 11]
```diff
@@ -85,11 +122,50 @@ export const useInventoryStore = defineStore('inventoryStore', {
...
       this.searchType = searchType
+      this.inventoryNodesDefaultFilter = {
+        ...this.inventoryNodesDefaultFilter,
+        searchType: searchType?.value
+      }
     },
     resetSelectedNode() {
       this.nodesSelected = []
+    },
+    setInventeriesByNodePage(page: number) {
```
Comment (synqotik) [inline ui/src/store/Views/inventoryStore.ts:135]: Rename to: `setInventoriesByNodePage`

File: ui/src/store/Views/inventoryStore.ts [hunk 12]
```diff
@@ -85,11 +122,50 @@ export const useInventoryStore = defineStore('inventoryStore', {
...
+      if (page !== Number(this.inventoryNodesPagination.page)) {
+        this.inventoryNodesPagination = {
+          ...this.inventoryNodesPagination,
+          page
+        }
+      }
+      this.init()
+    },
+    setInventeriesByNodePageSize(pageSize: number) {
```
Comment (synqotik) [inline ui/src/store/Views/inventoryStore.ts:144]: Rename to: `setInventoriesByNodePageSize`

File: ui/src/store/Views/inventoryStore.ts [hunk 13]
```diff
@@ -85,11 +122,50 @@ export const useInventoryStore = defineStore('inventoryStore', {
...
+        this.inventoryNodesPagination = {
+          ...this.inventoryNodesPagination,
+          page: 0, // always request first page on change
+          pageSize
+        }
+      }
+
+      this.init()
+    },
+    inventeriesByNodeSortChanged(sortObj: any) {
```
Comment (synqotik) [inline ui/src/store/Views/inventoryStore.ts:155]: Rename to: `inventoriesByNodeSortChanged`

File: ui/src/types/inventory.d.ts [hunk 14]
```diff
@@ -54,6 +70,13 @@ interface RawMetrics {
   }
 }
 
+interface InventoryItemsFilters {
```
Comment (synqotik) [inline ui/src/types/inventory.d.ts:73]: Rename to: ` InventoryItemFilters`

**Review by synqotik — CHANGES_REQUESTED** [review]
Looks good, just a few small name fixes needed

**Review by synqotik — APPROVED** [review]
LGTM

**Review by Shahbaz-dataq — APPROVED** [review]

**Pull request merged**